client/executor/wasmi/src/lib.rs

```diff
@@ -458,7 +458,11 @@ fn call_in_wasm_module(
 
 	// Write the call data
 	let offset = fec.allocate_memory(data.len() as u32)?;
-	fec.write_memory(offset, data)?;
```

Comment:
Please revert. 

If you read the source, you will be able to see that we drop everything here anyway.